README.md
MOOG_LTE
==========

MOOG_LTE

Authors
-------
- **Chris Sneden** (UT Austin)

Installation
------------

MOOG_LTE is a Fortran 77 (F77) code.  To install the latest version of the code, employ one of the OS-specific Makefiles (Makefile.*) and 
use the following command:

    make -f Makefile*

Accordingly, a Fortran compiler is necessary (for example, succesful MOOG_LTE compilation has been achieved with the ``gfortran`` compiler).  In addition, MOOG_LTE relies 
completely upon [SuperMongo](https://www.astro.princeton.edu/~rhl/sm/) for plotting/visualziation (latest release 2.4.1).	
    
Documentation
-------------
The "MOOG-LTE_Documentation_Nov2019.pdf" file is included in the distribution.  Future plans entail other forms of documentation.  


CHANGELOG.txt
File describes modifications made to the MOOG-LTE code since its release in Nov 2019 (as found on the C. Sneden website: https://www.as.utexas.edu/~chris/moog.html)

03 AUGUST 2023
------------

- (JS) Slight changes to text documentation of Partfn.f subroutine.
- (JS) Inmodel.new file is ignored and will eventually be removed from main directory
- (JS) Removal of PLOTVAL common block
- (JS) Modification of Correl.f to delete PLOTVALS common block
- (JS) Removal of common block file from distribution
- (JS) Deletion of the ISOPLOT driver call (and associated text) in the Moogsilent.f routine (note: slight additional departures btwn Moog and Moogsilent) 
- (JS) Deletion of the ISOPLOT driver call (and associated text) in the Params.f routine




Partfn.f

      subroutine partfn (atom,jmark)                                  
c******************************************************************************
c     This routine computes arrays or single partition functions;
c     It uses the data and formulation of Kurucz's ATLAS9 program.
c     However, the partition functions have been updated for several species,
c     and those are calculated in subroutine *partnew*.
c******************************************************************************

      implicit real*8 (a-h,o-z)
      include 'Atmos.com'
      include 'Linex.com'
      include 'Quants.com'


      iat = 10*nint(atom) - 1
      at = dfloat(iat)/10.
      iatom = nint(atom)


c*****compute partition functions for 4 ionization states of an element.
      do k=1,4
         iat = iat + 1
         at = dfloat(iat)/10.
         if (partflag(iatom,k) .gt. 0) then
            do i=1,ntau   
               u(jmark,k,i) = partnew(at,k,i)
            enddo
         else
            do i=1,ntau   
               u(jmark,k,i) = ucalc(at,i)   
            enddo
         endif
      enddo


      return
      end 






Moogsilent.f

      program moogsilent
c******************************************************************************
c     This is the main driver for the non-interactive version of MOOG.  
c     It reads the parameter file and sends MOOG to various controlling 
c     subroutines.  In this version of MOOG the parameter file must
c     be named "batch.par" (because the code cannot stop to ask the
c     user to name the parameter file)
c******************************************************************************

      include 'Atmos.com'
      include 'Pstuff.com'


c$$$$$$$$$$$$$$$$$$$$$$$$ USER SETUP AREA $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
c     in compiling MOOG, here the various machine-specific things are
c     declared.  First, define the directory where MOOG lives, in order to
c     be able to pull in auxiliary data files; executing 'make' will
c     generate a reminder of this
      write (moogpath,1001)
      moogpath = 
     .  '/Users/chris/CODES/moognov2019/'


c*****What kind of machine are you using?  Possible ones are:
c     "mac" = Intel-based Apple Mac
c     "pcl" = a PC or desktop running some standard linux like Redhat
c     "uni" = a machine running Unix, specifically Sun Solaris
      machine = "mac"


c*****for x11 terminal types, define the parameters of plotting windows;
c     set up an x11 screen geometry and placement that is good for spectrum
c     syntheses (long, but not tall); the user should play with the format
c     statements for particular machines.
      write (smt1,1018)
c     now do the same for line abundance trend plots (short but tall).
      write (smt2,1017)
c$$$$$$$$$$$$$$$$$$$$$$$ END OF USER SETUP $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$


c*****declare this to be the non-interactive version; variable "silent"
c     will be queried on all occasions that might call for user input;
c     DON'T CHANGE THIS VARIABLE; 
c     if silent = 'n', the normal interactive MOOG is run;
c     if silent = 'y', the non-interactive MOOG is run
      silent = 'y'


c*****invoke the overall starting routine
      control = '       '
      call begin


c*****use one of the standard driver routines
      if     (control .eq. 'synplot') then
         call plotit
      elseif (control .eq. 'synth  ') then
         call synth
      elseif (control .eq. 'cogsyn ') then
         call cogsyn  
      elseif (control .eq. 'blends ') then
         call blends  
      elseif (control .eq. 'abfind ') then
         call abfind
      elseif (control .eq. 'ewfind ') then
         call ewfind
      elseif (control .eq. 'cog    ') then
         call cog
      elseif (control .eq. 'calmod ') then
         call calmod
      elseif (control .eq. 'doflux ') then
         call doflux   
      elseif (control .eq. 'weedout') then
         call weedout  
      elseif (control .eq. 'gridsyn') then
         call gridsyn  
      elseif (control .eq. 'gridplo') then
         call gridplo  
      elseif (control .eq. 'binary ') then
         call binary
      elseif (control .eq. 'abpop  ') then
         call abpop
      elseif (control .eq. 'synpop ') then
         call synpop


c*****or, put in your own drivers in the form below....
      elseif (control .eq. 'mine  ') then
         call  mydriver 


c*****or else you are out of luck!
      else
         array = 'THIS IS NOT ONE OF THE DRIVERS. I QUIT!'
         istat = ivwrite (4,3,array,49)
         stop
      endif


c*****format statements
1001  format (60(' '))
1017  format ('x11 -bg black -title MOOGplot -geom 700x800+650+000')
1018  format ('x11 -bg black -title MOOGplot -geom 1200x400+20+450')


      end
      


Moog.f

      program moog
c******************************************************************************
c     This is the main driver for MOOG.  It reads the parameter
c     file and sends MOOG to various controlling subroutines.
c     This is the normal interactive version of the code; for batch
c     processing without user decisions, run MOOGSILENT instead.
c******************************************************************************

      include 'Atmos.com'
      include 'Pstuff.com'
      character yesno*1


c$$$$$$$$$$$$$$$$$$$$$$$$ USER SETUP AREA $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
c*****in compiling MOOG, here the various machine-specific things are 
c     declared.  First, define the directory where MOOG lives, in order to 
c     be able to pull in auxiliary data files; executing 'make' will 
c     generate a reminder of this necessity
      write (moogpath,1001)
      moogpath = 
     .  '/Users/jsobeck/moog2019_LTE'


c*****What kind of machine are you using?  Possible ones are:
c     "mac" = Intel-based Apple Mac 
c     "pcl" = a PC or desktop running some standard linux like Redhat
c     "uni" = a machine running Unix, specifically Sun Solaris
      machine = "mac"


c*****for x11 terminal types, define the parameters of plotting windows;
c     set up an x11 screen geometry and placement that is good for spectrum
c     syntheses (long, but not tall); the user should play with the format
c     statements for particular machines.
      write (smt1,1018)
c     now do the same for line abundance trend plots (short but tall).
      write (smt2,1017)
c$$$$$$$$$$$$$$$$$$$$$$$ END OF USER SETUP $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$


c*****declare this to be the normal interactive version; variable "silent"
c     will be queried on all occasions that might call for user input;
c     DON'T CHANGE THIS VARIABLE; 
c     if silent = 'n', the normal interactive MOOG is run;
c     if silent = 'y', the non-interactive MOOG is run
      silent = 'n'


c*****invoke the overall starting routine
1     control = '       '
      call begin


c*****use one of the standard driver routines
      if     (control .eq. 'synplot') then
         call plotit
      elseif (control .eq. 'synth  ') then
         call synth
      elseif (control .eq. 'cogsyn ') then
         call cogsyn  
      elseif (control .eq. 'blends ') then
         call blends  
      elseif (control .eq. 'abfind ') then
         call abfind
      elseif (control .eq. 'ewfind ') then
         call ewfind
      elseif (control .eq. 'cog    ') then
         call cog
      elseif (control .eq. 'calmod ') then
         call calmod
      elseif (control .eq. 'doflux ') then
         call doflux   
      elseif (control .eq. 'weedout') then
         call weedout  
      elseif (control .eq. 'gridsyn') then
         call gridsyn  
      elseif (control .eq. 'gridplo') then
         call gridplo  
      elseif (control .eq. 'binary ') then
         call binary
      elseif (control .eq. 'abpop  ') then
         call abpop
      elseif (control .eq. 'synpop ') then
         call synpop


c*****or, put in your own drivers in the form below....
      elseif (control .eq. 'mine  ') then
         call  mydriver 


c*****or else you are out of luck!
      else
         array = 'THIS IS NOT ONE OF THE DRIVERS.  TRY AGAIN (y/n)?'
         istat = ivwrite (4,3,array,49)
         istat = ivmove (3,1)
         read (*,*) yesno
         if (yesno .eq. 'y') then
            go to 1
         else
            call finish (0)
         endif
      endif


c*****format statements
1001  format (60(' '))
1003  format (22x,'MOOG IS CONTROLLED BY DRIVER ',a7)
1017  format ('x11 -bg black -title MOOGplot -geom 700x800+650+000')
1018  format ('x11 -bg black -title MOOGplot -geom 1200x350+20+450')


      end


      
